pso/src/main/scala/cilib/pso/PSO.scala

```diff
@@ -54,6 +54,187 @@ object PSO {
     if (b) updatePBest(p) else Step.point(p)
   }
 
+  def clamping[S](particle: Particle[S, Double])(
```

Comment:
This is too specific. We could do better here by ensuring that we don't specialize to `Double`, instead it would be better if we add a type-param and ensure that there is an `Order` instance instead so that the comparisons can be done, without caring what the type really is.